src/main/java/nextstep/sessions/domain/data/Session.java

```diff
@@ -24,6 +40,10 @@ public Session(SessionInfo sessionInfo, Registrations registrations) {
         this.registrations = registrations;
     }
 
+    public Session with(List<Registration> registrations) {
+        return new Session(sessionInfo, registrations);
+    }
+
     public Registration registration(NsUser user, Payment payment) {
         sessionInfo.validateEnrollment(registrations.size(), payment);
         registrations.validateDuplicateEnrollment(user);
```

Comment:
수강신청에 대한 책임을 하나의 객에게 모두 위임하는 구조도 고려해 보시면 좋을 것 같아요
```suggestion
    public Registration registration(NsUser user, Payment payment) {
        return enrollment.enroll(user, payment);
```

Reply:
이런식으로 멤버변수 개수를 줄일 수 있군요!
1) 한 객체의 책임이 많아지면 책임을 분담할 새로운 객체를 생성하고
2) 기존 객체는 관리하던 필요한 정보(멤버변수)를 담아 새로운 객체를 리턴하는 메서드를 만든다.
3) 해당 메서드를 통해 새로운 객체를 생성하고, 새로운 객체에서는 전달 받은 멤버변수를 활용하여 메서드로 역할을 수행한다.

Session이 enrollment를 멤버변수로 가질 필요가 없다고 생각해서
Session에게 검증에 필요한 수강목록을 인자로주면 Enrollment 객체를 리턴하도록 했습니다.

Enrollment에서는 enroll() 메서드로 검증을 수행하도록 했습니다.
user, payment 정보를 가지고 검증하고 이상 없으면 해당 정보를 담아 Registration을 리턴하는 로직이 있었습니다.
Registration에 user, payment 정보를 미리 담아 인자로 주고, enroll(registration) 메서드가 동작하도록 했습니다.
이렇게 하면 user, payment 정보를 별도로 주지 않아도 됩니다.

Reply:
Session과 Enrollment가 SessionInfo라는 멤버변수를 동일하게 갖게 되는데... 괜찮은 상황인가요 ..?